test/mocks/MockBadSubscribers.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.20;
+
+import {ISubscriber} from "../../src/interfaces/ISubscriber.sol";
+import {PositionConfig} from "../../src/libraries/PositionConfig.sol";
+import {PositionManager} from "../../src/PositionManager.sol";
+
+/// @notice A subscriber contract that returns values from the subscriber entrypoints
+contract MockReturnDataSubscriber is ISubscriber {
+    PositionManager posm;
+
+    uint256 public notifySubscribeCount;
+    uint256 public notifyUnsubscribeCount;
+    uint256 public notifyModifyLiquidityCount;
+    uint256 public notifyTransferCount;
+
+    error NotAuthorizedNotifer(address sender);
+
+    error NotImplemented();
+
+    uint256 memPtr;
+
+    constructor(PositionManager _posm) {
+        posm = _posm;
+    }
+
+    modifier onlyByPosm() {
+        if (msg.sender != address(posm)) revert NotAuthorizedNotifer(msg.sender);
+        _;
+    }
+
+    function notifySubscribe(uint256 tokenId, PositionConfig memory config) external onlyByPosm {
+        notifySubscribeCount++;
+    }
+
+    function notifyUnsubscribe(uint256 tokenId, PositionConfig memory config) external onlyByPosm {
+        notifyUnsubscribeCount++;
+        uint256 _memPtr = memPtr;
```

Comment:
isnt memPtr always 0? i dont see a way it can be set?

Reply:
can you explain what this subscriber does thats bad?

Reply:
notify unsubscribe returns too much data

i call setReturnDataSize to a large size and so the subscriber's notifyUnsubscribeCount does not get set bc it runs OOG